pallets/file-system/src/tests.rs

```diff
@@ -3972,6 +3972,593 @@ mod compute_threshold {
+                size,
+                fingerprint,
+            );
+
+            let new_root =
+				<<Test as crate::Config>::Providers as shp_traits::ReadProvidersInterface>::get_root(
+					bsp_id,
+				)
+					.unwrap();
+
+            // Assert that the correct event was deposited
+            System::assert_last_event(
+                Event::BspConfirmedStoring {
+                    who: bsp_account_id.clone(),
+                    bsp_id,
+                    file_keys: BoundedVec::try_from(vec![file_key]).unwrap(),
+                    new_root,
+                }
+                .into(),
+            );
+
+            // Assert that the proving cycle was initialised for this BSP.
+            let last_tick_provider_submitted_proof =
+                LastTickProviderSubmittedAProofFor::<Test>::get(&bsp_id).unwrap();
+            assert_eq!(last_tick_provider_submitted_proof, tick_when_confirming);
+
+            // Assert that the correct event was deposited.
+            System::assert_has_event(
+                Event::BspChallengeCycleInitialised {
+                    who: bsp_account_id,
+                    bsp_id,
+                }
+                .into(),
+            );
+
+            // Assert that the capacity used by the BSP was updated
+            assert_eq!(
+                pallet_storage_providers::BackupStorageProviders::<Test>::get(bsp_id)
+                    .expect("BSP should exist in storage")
+                    .capacity_used,
+                size
+            );
+
+            // Now that the BSP has confirmed storing, we can simulate the user being insolvent
+            // and the BSP stopping storing for the user.
+            // Try to stop storing for the insolvent user.
+            assert_ok!(FileSystem::stop_storing_for_insolvent_user(
+                bsp_signed.clone(),
+                file_key,
+                bucket_id,
+                location.clone(),
+                owner_account_id.clone(),
+                fingerprint,
+                size,
+                CompactProof {
+                    encoded_nodes: vec![file_key.as_ref().to_vec()],
+                },
+            ));
+
+            // Assert that the correct event was deposited
+            System::assert_last_event(
+                Event::SpStopStoringInsolventUser {
+                    sp_id: bsp_id,
+                    file_key,
+                    owner: owner_account_id,
+                    location,
+                    new_root,
+                }
+                .into(),
+            );
+
+            // Assert that the capacity used by the BSP was updated
+            assert_eq!(
+                pallet_storage_providers::BackupStorageProviders::<Test>::get(bsp_id)
+                    .expect("BSP should exist in storage")
+                    .capacity_used,
+                0
+            );
+        });
+    }
+
+    // TODO: Not sure if needed, since we might not store any MSP info in the future
```

Comment:
Yes, we still store the bucket roots, we just don't require proofs from them regularly.

Reply:
Great, when the `msp_accept_storage_request` PR is merged I'll add this test using that.